tests/store/lineage/test_downstream_lineage.py

```diff
@@ -0,0 +1,104 @@
+import base64
+from unittest import mock
+
+import pytest
+
+import mlflow
+from mlflow.protos.databricks_uc_registry_messages_pb2 import (
+    Entity,
+    Job,
+    LineageHeaderInfo,
+    Notebook,
+)
+from mlflow.store._unity_catalog.lineage.constants import _DATABRICKS_LINEAGE_ID_HEADER
+from mlflow.store._unity_catalog.registry.rest_store import UcModelRegistryStore
+from mlflow.store.artifact.s3_artifact_repo import S3ArtifactRepository
+from mlflow.store.artifact.unity_catalog_models_artifact_repo import (
+    UnityCatalogModelsArtifactRepository,
+)
+from mlflow.utils.proto_json_utils import message_to_json
+
+
+class SimpleModel(mlflow.pyfunc.PythonModel):
+    def predict(self, _, model_input):
+        return model_input.applymap(lambda x: x * 2)
+
+
+@pytest.fixture
+def store(mock_databricks_uc_host_creds):
+    with mock.patch("mlflow.utils.databricks_utils.get_config"):
+        yield UcModelRegistryStore(store_uri="databricks-uc", tracking_uri="databricks")
+
+
+def lineage_header_info_to_extra_headers(lineage_header_info):
+    extra_headers = {}
+    if lineage_header_info:
+        header_json = message_to_json(lineage_header_info)
+        header_base64 = base64.b64encode(header_json.encode())
+        extra_headers[_DATABRICKS_LINEAGE_ID_HEADER] = header_base64
+    return extra_headers
+
+
+@pytest.mark.parametrize(
+    ("notebook_id", "job_id"),
+    [
+        (None, None),
+        (1234, None),
+        (None, 5678),
+        (1234, 5678),
+    ],
+)
+def test_downstream_notebook_job_lineage(tmp_path, notebook_id, job_id):
+    model_dir = str(tmp_path.joinpath("mymodel"))
+    model_name = "mycatalog.myschema.mymodel"
+    model_uri = f"models:/{model_name}/1"
+
+    mock_artifact_repo = mock.MagicMock(autospec=S3ArtifactRepository)
+    mock_artifact_repo.download_artifacts.return_value = model_dir
+
+    entity_list = []
+    if notebook_id:
+        notebook_entity = Notebook(id=str(notebook_id))
+        entity_list.append(Entity(notebook=notebook_entity))
+
+    if job_id:
+        job_entity = Job(id=str(5678))
+        entity_list.append(Entity(job=job_entity))
+
+    expected_lineage_header_info = LineageHeaderInfo(entities=entity_list) if entity_list else None
+
+    with mock.patch(
+            "mlflow.utils.databricks_utils.get_config"
+    ), mock.patch(
+        "mlflow.utils.databricks_utils.is_in_databricks_notebook",
+        return_value=True,
```

Comment:
can we add another test for when the is_in_databricks_(notebook|job) are both false assert that the lineage info is None?

Reply:
thanks for reviewing. 
updated pr with test you suggested.